include/wil/common.h

```diff
@@ -325,8 +325,12 @@ Three exception modes are available:
 #define WIL_HAS_CXX_17 0
 #endif
 
-// Until we'll have C++17 enabled in our code base, we're falling back to SAL
+#if __has_cpp_attribute(nodiscard)

```

Comment:
Does this need to be guarded with a check for the presence of the `__has_cpp_attribute` doodad?  I see things like:

```c
#ifdef __has_cpp_attribute
#if __has_cpp_attribute(foo)
#else
#endif
#endif
```